core/app/services/uffizzi_core/account_service.rb

```diff
@@ -10,6 +10,14 @@ def create_credential(credential)
       end
     end
 
+    def update_credentials(credentials)
+      credentials.account.projects.active.each do |project|
+        project.deployments.active.each do |deployment|
```

Comment:
Let's fetch all the necessary deployments in an SQL query instead of nested cycles